component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java

```diff
@@ -0,0 +1,236 @@
+
+                result = this.pageService.findPages(offset, limit, siteType, null, null, null);
+                if (result != null && result.getSize() > 0) {
+                    Iterator<PageContext> iterator = result.iterator();
+                    while (iterator.hasNext()) {
+                        PageContext page = iterator.next();
+
+                        LOG.info("Remove gadgets in page: " + page.getKey().format());
+                        this.migratePage(page);
+                        LOG.info("Migrate done");
+                    }
+                    offset += limit;
+                }
+            } catch (Exception e) {
+                LOG.error("Error during migrate gadgets: " + e.getMessage(), e);
+            } finally {
+                RequestLifeCycle.end();
+            }
+
+        } while (result != null && result.getSize() > 0);
+    }
+
+    protected void migratePage(PageContext page) throws Exception {
+        PageData pageData = this.modelDataStorage.getPage(PageKey.create(page.getKey().format()));
+        pageData = migratePage(pageData);
+        this.modelDataStorage.save(pageData);
+    }
+
+    protected PageData migratePage(PageData pageData) throws Exception {
+        List<ComponentData> children = this.migrateComponents(pageData.getChildren());
+        return new PageData(
+                pageData.getStorageId(),
+                pageData.getId(),
+                pageData.getName(),
+                pageData.getIcon(),
+                pageData.getTemplate(),
+                pageData.getFactoryId(),
+                pageData.getTitle(),
+                pageData.getDescription(),
+                pageData.getWidth(),
+                pageData.getHeight(),
+                pageData.getAccessPermissions(),
+                children,
+                pageData.getOwnerType(),
+                pageData.getOwnerId(),
+                pageData.getEditPermission(),
+                pageData.isShowMaxWindow(),
+                pageData.getMoveAppsPermissions(),
+                pageData.getMoveContainersPermissions()
+        );
+    }
+
+    protected List<ComponentData> migrateComponents(List<ComponentData> components) throws Exception {
+        List<ComponentData> children = new ArrayList<>();
+        for(ComponentData ele : components) {
+            ComponentData migrated = this.migrateComponent(ele);
+            if (migrated != null) {
+                children.add(migrated);
+            }
+        }
+        return children;
+    }
+
+    protected ComponentData migrateComponent(ComponentData component) throws Exception {
+        if (component instanceof ApplicationData) {
+            ApplicationData<?> app = (ApplicationData<?>)component;
+            ApplicationState state = app.getState();
+            String contentId = this.modelDataStorage.getId(state);
+
+            if (app.getType() == ApplicationType.GADGET) {
+                LOG.info("Migrate for gadget: " + contentId);
```

Comment:
We should add the name of the page IMO.
```suggestion
                LOG.info("Gadget " + contentId + " found in page " + page);
```